scoop.lua

```diff
@@ -0,0 +1,338 @@
+-- -*- coding: utf-8 -*-
+-- preamble: common routines
+
+local JSON = require("JSON")
+
+local matchers = require("matchers")
+local path = require("path")
+local w = require("tables").wrap
+local concat = require("funclib").concat
+
+local parser = clink.arg.new_parser
+
+local function scoop_folder()
+    local folder = os.getenv("SCOOP")
+
+    if not folder then
+        folder = os.getenv("home") .. "\\scoop"
+    end
+
+    return folder
+end
+
+local function scoop_global_folder()
+    local folder = os.getenv("SCOOP_GLOBAL")
+
+    if not folder then
+        folder = os.getenv("ProgramData") .. "\\scoop"
+    end
+
+    return folder
+end
+
+local function scoop_load_config() -- luacheck: no unused args
+    local file = io.open(os.getenv("home") .. "\\.scoop")
+    -- If there is no such file, then close handle and return
+    if file == nil then
+        return w()
+    end
+
+    -- Read the whole file contents
+    local contents = file:read("*a")
+    file:close()
+
+    -- strip UTF-8-BOM
+    local utf8_len = contents:len()
+    local pat_start = string.find(contents, "{")[0]
```

Comment:
Not working for me. 
I've changed it to:
`local pat_start, _ = string.find(contents, "{")`
to make it work on my pc

Reply:
I will check later and fix